src/classes/TDTM_Config.cls

```diff
@@ -50,33 +50,65 @@ public class TDTM_Config {
 
     /********************************************************************************************************
     * @description Returns the TDTM configuration. If no configuration is currently set, it will return
-    * the default configuration without setting it. The configuration returned has not yet been inserted in the
-    * database. It can be manipulated, if necessary, then set.
+    * the default configuration without setting it. If the default configuration is the one returned,
+    * it has not yet been inserted in the database. It can be manipulated, if necessary, then set.
     * If no configuration exists, the application will insert the defaults when any of the triggers run.
     * @return List<TDTM_Global_API.TdtmToken> List of trigger handler tokens with the default values.
     *********************************************************************************************************/
     public static List<TDTM_Global_API.TdtmToken> getTdtmConfig() {
         if(tdtmTokensCache != null && tdtmTokensCache.size() > 0) {
             return tdtmTokensCache;
         }
-        List<TDTM_Global_API.TdtmToken> tokens = new List<TDTM_Global_API.TdtmToken>();
+
+        Boolean useAdv = false;
+        String apiHedaClass = '';
+        //Check if Advancement is installed and enabled.
+        try {
+            List<SObject> advSettings = Database.query('select Use_Advancement__c, API_HEDA_Class__c from Advancement_Setting__mdt');
```

Comment:
won't these fields and metadata object from the advancement package need to be namespaced?